js_modules/dagster-ui/packages/ui-core/src/jobs/JobsPageContent.tsx

```diff
@@ -45,18 +51,28 @@ export const JobsPageContent = () => {
       notifyOnNetworkStatusChange: true,
     },
   );
-  useBlockTraceOnQueryResult(queryResultOverview, 'OverviewJobsQuery');
   const {data, loading} = queryResultOverview;
 
   const refreshState = useQueryRefreshAtInterval(queryResultOverview, FIFTEEN_SECONDS);
 
   // Batch up the data and bucket by repo.
   const repoBuckets = useMemo(() => {
+    const cachedEntries = Object.values(cachedData).filter(
+      (location): location is Extract<typeof location, {__typename: 'WorkspaceLocationEntry'}> =>
+        location.__typename === 'WorkspaceLocationEntry',
+    );
+    const workspaceOrError = data?.workspaceOrError;
+    const entries =
+      workspaceOrError?.__typename === 'Workspace'
+        ? workspaceOrError.locationEntries
+        : cachedEntries;
     const visibleKeys = visibleRepoKeys(visibleRepos);
-    return buildBuckets(data).filter(({repoAddress}) =>
+    return buildBuckets(entries).filter(({repoAddress}) =>
       visibleKeys.has(repoAddressAsHumanString(repoAddress)),
     );
-  }, [data, visibleRepos]);
+  }, [cachedData, data, visibleRepos]);
+
+  useBlockTraceUntilTrue('WorkspaceJobs', !!data || !workspaceLoading);
```

Comment:
I think I should probably keep the existing blocking instead since the workspace cache would have stale stateful data and down the line we're going to split out the stateful stuff from the workspace which would make that change seem like a regression, I'll revert this.